### PR TITLE
71 in memory fixed

### LINKED ISSUE #71: Ensure memory pinning is happening properly
**Is your feature request related to a problem? Please describe.**
Currently, memory pinning doesn't work for CPU data

**Describe the solution you'd like**
Implement solution based on https://github.com/fepegar/torchio/discussions/568#discussioncomment-859027

**Describe alternatives you've considered**
N.A.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #71 

## Proposed Changes

- `in_memory` is now added for cpu and gpu separately
- once `in_memory` is selected, the data gets loaded onto the RAM and is lazy loading gets disabled
- this significantly reduces the epoch time (based on the I/O capabilities of the machine you are running GaNDLF on)
- version updated for development
- added evaluation of string values of certain parameters

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GANDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`

### CODE REVIEW

File: GANDLF/parseConfig.py [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
 parameter_defaults = {
     "weighted_loss": False, # whether weighted loss is to be used or not
     "verbose": False, # general application verbosity
     "q_verbose": False, # queue construction verbosity
     "medcam_enabled": False, # interpretability via medcam
     "save_output": False, # save outputs during validation/testing
-    "in_memory": False, # pin data to gpu memory
-    "patch_sampler": "uniform", # type of sampling strategy
+    "in_memory": False, # pin data to cpu memory
+    "in_memory_gpu": False, # pin data to gpu memory
```
Comment (Geeks-Sid) [inline GANDLF/parseConfig.py:11]: I am pretty sure we are not using DALI or any in_memory_gpu supporting libraries.

File: GANDLF/training_loop.py [hunk 2]
```diff
@@ -480,16 +480,16 @@ def training_loop(
         training_data_for_torch,
         batch_size=params["batch_size"],
         shuffle=True,
-        pin_memory=params["in_memory"],
+        pin_memory=params["in_memory_gpu"],
```
Comment (Geeks-Sid) [inline GANDLF/training_loop.py:483]: https://stackoverflow.com/a/55564072/6784556

Can we take a look into this?

**Review by Geeks-Sid — CHANGES_REQUESTED** [review]

File: GANDLF/parseConfig.py (hunk 1 above)
Comment (sarthakpati) [reply GANDLF/parseConfig.py:11]: Actually, this is an option for this: https://pytorch.org/docs/stable/data.html#memory-pinning

File: GANDLF/training_loop.py (hunk 2 above)
Comment (sarthakpati) [reply GANDLF/training_loop.py:483]: Sure, that makes sense. This is the reason why I put `in_memory_gpu` as a separate option compared to `in_memory`.

**Review by Geeks-Sid — APPROVED** [review]

**Pull request merged**